packages/time-picker/test/validation.test.js

```diff
@@ -68,37 +68,57 @@ describe('validation', () => {
   });
 
   describe('basic', () => {
-    let validateSpy;
+    let validateSpy, changeSpy, input;
 
     beforeEach(() => {
       timePicker = fixtureSync(`<vaadin-time-picker></vaadin-time-picker>`);
       validateSpy = sinon.spy(timePicker, 'validate');
+      changeSpy = sinon.spy();
+      timePicker.addEventListener('change', changeSpy);
+      input = timePicker.inputElement;
     });
 
     it('should pass validation by default', () => {
       expect(timePicker.checkValidity()).to.be.true;
       expect(timePicker.validate()).to.be.true;
     });
 
-    it('should not validate on value input', () => {
+    it('should not validate on user input', () => {
       setInputValue(timePicker, '12:00');
-      expect(validateSpy.calledOnce).to.be.false;
+      expect(validateSpy.called).to.be.false;
+    });
+
+    it('should validate on blur', () => {
+      input.focus();
+      input.blur();
+      expect(validateSpy.calledOnce).to.be.true;
     });
 
-    it('should validate on value commit', () => {
+    it('should validate before change event on blur', () => {
+      input.focus();
       setInputValue(timePicker, '12:00');
-      enter(timePicker.inputElement);
+      input.blur();
+      expect(changeSpy.calledOnce).to.be.true;
       expect(validateSpy.calledOnce).to.be.true;
+      expect(validateSpy.calledBefore(changeSpy)).to.be.true;
     });
 
-    it('should validate on input focusout', () => {
-      focusout(timePicker.inputElement);
+    it('should validate before change event on Enter', () => {
+      setInputValue(timePicker, '12:00');
+      enter(timePicker.inputElement);
+      expect(changeSpy.calledOnce).to.be.true;
       expect(validateSpy.calledOnce).to.be.true;
+      expect(validateSpy.calledBefore(changeSpy)).to.be.true;
     });
 
-    it('should be possible to force invalid status', () => {
```

Comment:
It's already covered by this snapshot test:

https://github.com/vaadin/web-components/blob/6c099fe9808eb62873adcda78e5da0b3da2d486d/packages/time-picker/test/dom/time-picker.test.js#L30-L35